extra/1-factorial.js

```diff
@@ -9,7 +9,13 @@
 */
 
 function factorial(input) {
-  // TODO
+  let result = input;
+  while (input > 1) {
+    input--; 
+    result *= input;  
+    
+  }
+  return result;
 }
```

Comment:
You used an iterative approach, it is very good and it works and I liked it, but you can also do it like below: 
```js
function factorial(input) {
  if (input < 2) {
    return 1;
  } else {
    return input * factorial(input - 1);
  }
}
```
This approach is called the recursive approach! You will learn more about it in the future.

Reply:
Oh this looks very interesting! I will be reading about recursive approach later today after doing the homework. Thank you for pointing this out! 👍 :)